### PR TITLE
Fix issues reported in ZD13606

### PR DESCRIPTION
**Pull request by julek-wolfssl**
- Ticket failure should result in a regular handshake
- Fix `BioReceive` for closed connection
  -The pending check was forcing a `WOLFSSL_CBIO_ERR_WANT_WRITE` return even though the underlying socket was closed and `WOLFSSL_BIO_FLAG_READ|WOLFSSL_BIO_FLAG_RETRY` was not set. The `wolfSSL_BIO_ctrl_pending(ssl->biord) == 0` is old and I can't find a reason to keep checking it. I left it just in the case where there is output data pending.

ZD 13606

### CODE REVIEW

**Review by dgarske — APPROVED** [review]
Over to @SparkiDev for another review on the ticket change. Makes sense to me, but I'd like him to confirm.

File: src/wolfio.c [hunk 1]
```diff
@@ -169,15 +169,17 @@ int BioReceive(WOLFSSL* ssl, char* buf, int sz, void* ctx)
 
     recvd = wolfSSL_BIO_read(ssl->biord, buf, sz);
     if (recvd <= 0) {
-        if (wolfSSL_BIO_supports_pending(ssl->biord) &&
+        if (/* ssl->biowr->wrIdx is checked for Bind9 */
+            ssl->biowr != NULL && ssl->biowr->type == WOLFSSL_BIO_BIO &&
```
Comment (SparkiDev) [inline src/wolfio.c:173]: Add wolfSSL_BIO_method_type() to do this line.

File: src/wolfio.c [hunk 2]
```diff
@@ -169,15 +169,17 @@ int BioReceive(WOLFSSL* ssl, char* buf, int sz, void* ctx)
...
+            ssl->biowr->wrIdx != 0 &&
```
Comment (SparkiDev) [inline src/wolfio.c:174]: Is this a check for pending write?
I believe the OpenSSL API is BIO_wpending(). So add wolfSSL_BIO_wpending().

File: src/internal.c [hunk 3]
```diff
@@ -30757,7 +30757,9 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
                                     et->enc_ticket, inLen, &outLen,
                                     ssl->ctx->ticketEncCtx);
         }
-        if (ret == WOLFSSL_TICKET_RET_FATAL || ret < 0) return ret;
+        if (ret == WOLFSSL_TICKET_RET_FATAL)
+            ret = WOLFSSL_TICKET_RET_REJECT;
+        if (ret < 0) return ret;
```
Comment (SparkiDev) [inline src/internal.c:30762]: Move return statement to next line.
Thanks.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/internal.c (hunk 3 above)
Comment (julek-wolfssl) [reply src/internal.c:30762]: Moved.

File: src/ssl.c [hunk 4]
```diff
@@ -17590,6 +17590,11 @@ int wolfSSL_set_compression(WOLFSSL* ssl)
             BIO_set_retry_read(wr);
         }
     }
+
+    int wolfSSL_BIO_method_type(const WOLFSSL_BIO *b)
```
Comment (SparkiDev) [inline src/ssl.c:17594]: I've moved all the wolfSSL_BIO functions to bio.c.
Move it there too. Thanks!

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 4 above)
Comment (julek-wolfssl) [reply src/ssl.c]: Moved to `bio.c`.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**